src/stirling/source_connectors/socket_tracer/protocols/mqtt/parse.cc

```diff
@@ -0,0 +1,774 @@
+  AuthenticationData = 0x16,
+  RequestProblemInformation = 0x17,
+  WillDelayInterval = 0x18,
+  RequestResponseInformation = 0x19,
+  ResponseInformation = 0x1A,
+  ServerReference = 0x1C,
+  ReasonString = 0x1F,
+  ReceiveMaximum = 0x21,
+  TopicAliasMaximum = 0x22,
+  TopicAlias = 0x23,
+  MaximumQos = 0x24,
+  RetainAvailable = 0x25,
+  UserProperty = 0x26,
+  MaximumPacketSize = 0x27,
+  WildcardSubscriptionAvailable = 0x28,
+  SubscriptionIdentifiersAvailable = 0x29,
+  SharedSubscriptionAvailable = 0x2A,
+  Invalid = 0xFF
+};
+
+static inline MqttControlPacketType GetControlPacketType(uint8_t control_packet_type_code) {
+  if (control_packet_type_code == static_cast<uint8_t>(MqttControlPacketType::PUBLISH)) {
+    return MqttControlPacketType::PUBLISH;
+  }
+
+  switch (control_packet_type_code) {
+    case static_cast<uint8_t>(MqttControlPacketType::CONNECT):
+      return MqttControlPacketType::CONNECT;
+    case static_cast<uint8_t>(MqttControlPacketType::CONNACK):
+      return MqttControlPacketType::CONNACK;
+    case static_cast<uint8_t>(MqttControlPacketType::PUBACK):
+      return MqttControlPacketType::PUBACK;
+    case static_cast<uint8_t>(MqttControlPacketType::PUBREC):
+      return MqttControlPacketType::PUBREC;
+    case static_cast<uint8_t>(MqttControlPacketType::PUBREL):
+      return MqttControlPacketType::PUBREL;
+    case static_cast<uint8_t>(MqttControlPacketType::PUBCOMP):
+      return MqttControlPacketType::PUBCOMP;
+    case static_cast<uint8_t>(MqttControlPacketType::SUBSCRIBE):
+      return MqttControlPacketType::SUBSCRIBE;
+    case static_cast<uint8_t>(MqttControlPacketType::SUBACK):
+      return MqttControlPacketType::SUBACK;
+    case static_cast<uint8_t>(MqttControlPacketType::UNSUBSCRIBE):
+      return MqttControlPacketType::UNSUBSCRIBE;
+    case static_cast<uint8_t>(MqttControlPacketType::UNSUBACK):
+      return MqttControlPacketType::UNSUBACK;
+    case static_cast<uint8_t>(MqttControlPacketType::PINGREQ):
+      return MqttControlPacketType::PINGREQ;
+    case static_cast<uint8_t>(MqttControlPacketType::PINGRESP):
+      return MqttControlPacketType::PINGRESP;
+    case static_cast<uint8_t>(MqttControlPacketType::DISCONNECT):
+      return MqttControlPacketType::DISCONNECT;
+    default:
+      return MqttControlPacketType::INVALID;
+  }
+}
+
+static inline StatusOr<size_t> VariableEncodingNumBytes(unsigned long integer) {
+  if (integer >= 268435456) {
+    return error::ResourceUnavailable("Maximum number of bytes exceeded for variable encoding.");
+  }
+
+  if (integer < 128) {
+    return 1;
+  } else if (integer < 16384) {
+    return 2;
+  } else if (integer < 2097152) {
+    return 3;
+  }
+  return 4;
+}
+
+ParseState ParseProperties(Message* result, BinaryDecoder* decoder, size_t& properties_length) {
+  uint8_t property_code;
+  while (properties_length > 0) {
+    // Extracting the property code
+    PX_ASSIGN_OR_RETURN_INVALID(property_code, decoder->ExtractBEInt<uint8_t>());
```

Comment:
In order to avoid all the `static_cast<uint8_t>` cases, we could use `magic_enum::enum_cast`.

```
uint8_t property_code;
PX_ASSIGN_OR_RETURN_INVALID(property_code, decoder->ExtractBEInt<uint8_t>());
auto property = magic_enum::enum_cast<PropertyCode>(property_code);

if (!property.has_value()) {
  return ParseState::kInvalid;
}

switch (property.value()) {
    case PropertyCode::PayloadFormatIndicator: {
...

```

Given there are quite a few cases, I think it will help with the readability.